cpp/include/cudf/null_mask.hpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/types.hpp>
+
+#include <rmm/device_buffer.hpp>
+#include <rmm/mr/device_memory_resource.hpp>
+
+namespace cudf {
+
+/**---------------------------------------------------------------------------*
+ * @brief Controls the allocation/initialization of a null mask.
+ *---------------------------------------------------------------------------**/
+enum mask_state {
+  UNALLOCATED,    ///< Null mask not allocated, (all elements are valid)
+  UNINITIALIZED,  ///< Null mask allocated, but not initialized
```

Comment:
How are we going to track if the mask is initialized or not? It's not clear to me why we need this state.

Reply:
This state is useful if you want to allocate a new column and you want to fill all of the mask bits yourself and would like to avoid the overhead of initializing the bitmask in the first place.